documentation/docs/extensibility.md

```diff
@@ -1,86 +1,199 @@
+Before building extensions, please make sure that there is no alternative which works better for you.
 
-!!! note "Cloud SDK Pipeline"
-    If you use the Cloud SDK Pipeline, the folder is named `pipeline/extensions/` (without the dot). For more information, please refer to [the Cloud SDK Pipeline documentation](https://github.com/SAP/cloud-s4-sdk-pipeline/blob/master/doc/pipeline/extensibility.md).
+Options for extensibility, in the order in which we recommend considering them:
 
-  The pipeline template checks if such a file exists and executes it, if present.
-  A parameter that contains the following keys is passed to the extension:
+## 1) Extend individual stages
 
-  * `script`: defines the global script environment of the Jenkinsfile run. This makes sure that the correct configuration environment can be passed to project "Piper" steps and also allows access to for example the `commonPipelineEnvironment`.
-  * `originalStage`: this will allow you to execute the "original" stage at any place in your script. If omitting a call to `originalStage()` only your code will be executed instead.
-  * `stageName`: name of the current stage
-  * `config`: configuration of the stage (including all defaults)
+In this option, you use the centrally maintained pipeline, but can change individual stages if required.
 
-  Here a simple example for such an extension:
+To do so, create a file called `<StageName>.groovy` (for example, `Acceptance.groovy` or `lint.groovy`) in `.pipeline/extensions/` in your application's source code repository.
 
-  ``` groovy
-  void call(Map params) {
-      //access stage name
-      echo "Start - Extension for stage: ${params.stageName}"
+For this, you need to know the technical identifiers for stage names.
 
-      //access config
-      echo "Current stage config: ${params.config}"
+* For the general purpose pipeline, you can find them in [the pipeline source file](https://github.com/SAP/jenkins-library/blob/master/vars/piperPipeline.groovy).
+* For SAP Cloud SDK Pipeline, you can find them in [this GitHub search query](https://github.com/SAP/cloud-s4-sdk-pipeline-lib/search?q=%22def+stageName+%3D%22).
 
-      //execute original stage as defined in the template
-      params.originalStage()
+The centrally maintained pipeline checks if such a file exists and executes it, if present.
+A parameter of type `Map` that contains the following keys is passed to the extension:
 
-      //access overall pipeline script object
-      echo "Branch: ${params.script.commonPipelineEnvironment.gitBranch}"
+* `script`: defines the global script environment of the `Jenkinsfile` run. This makes sure that the correct configuration environment can be passed to project "Piper" steps and allows access to for example the `commonPipelineEnvironment`.
+* `originalStage`: this will allow you to execute the "original" stage at any place in your script. If omitting a call to `originalStage()` only your code will be executed instead.
```

Comment:
```suggestion
* `originalStage`: this will allow you to execute the "original" stage at any place in your script. If omitting a call to `originalStage()` only your code will be executed.
```